pkgs/development/lisp-modules-new/imported.nix

```diff
@@ -10,14 +10,28 @@
 # `system`, not `asd`, so otherwise `system` would not be loadable
 # without building and loading `asd` first.
 #
-let createAsd = { url, sha256, asd, system }:
-   let
-     src = fetchzip { inherit url sha256; };
-   in runCommand "source" {} ''
-      mkdir -pv $out
-      cp -r ${src}/* $out
-      find $out -name "${asd}.asd" | while read f; do mv -fv $f $(dirname $f)/${system}.asd || true; done
-  '';
+let
+  createAsd = { url, sha256, asd, system }:
+    let
+      src = fetchzip { inherit url sha256; };
+    in
+      # This insane awk script strips out any defsystem that is not for this
+      # specific system. Possible areas of improvement:
+      # - case insensitivity
+      # - match e.g. (asdf/parse-defsystem:defsystem
+      # To fix as problems arise.
+      # The '"'"'" in particular is painful--it’s a single quote followed by a
+      # double quote, so '", but with shell escaping.
+      runCommand "source" {} ''
+        mkdir -pv $out
+        cp -r ${src}/* $out
+        find $out -name "${asd}.asd" | while read f; do
+          temp=$(mktemp)
+          awk '(/^$/ || /^[^ \t]/){f=0} (!/^\(asdf:defsystem [#:'"'"'"]*${system}"?$/ && /^\(asdf:defsystem/){f=1} !f{ print $0 }' "$f" > "$temp"
```

Comment:
is awk guaranteed to be available at this point? Is it part of the default nix build env?